astropy/nddata/mixins/ndarithmetic.py

```
@@ -23,212 +168,638 @@ class NDArithmeticMixin(object):
-            propagate_uncertainties = "propagate_subtract"
+        # ok, not really arithmetics but we need to check which wcs makes sense
+        # for the result and this is an ideal place to compare the two WCS,
+        # too.
+
+        if self.wcs is None and operand.wcs is None:
+            # None has any wcs information so the result should have None, no
+            # need to compare the WCS here.
+            return None
+        elif self.wcs is None:
+            # Only the second operand had WCS information so take this and
+            # issue a warning if comparison of WCS is requested.
+            if compare_wcs:
+                log.info("Only the second operand had some WCS information.")
+            return deepcopy(operand.wcs)
+        elif operand.wcs is None:
+            # Same as last case but reversed
+            if compare_wcs:
+                log.info("Only the first operand had some WCS information.")
+            return deepcopy(self.wcs)
         else:
-            propagate_uncertainties = None
-        return self._arithmetic(
-            operand, propagate_uncertainties, "subtraction", np.subtract)
+            # Both have WCS information. If requested compare the WCS and
+            # issue a warning if they are not equal
+            # TODO: Currently (astropy 1.04) this is a bit tricky because no
+            # two astropy.wcs.WCS are really identical even if they are copied
+            # so there is no reason to raise an Exception here, but issue the
+            # warning nevertheless. Really unfortunate... I currently worked
+            # around it so the warning differs if both are astropy.wcs.WCS
+            # instances.
+            if compare_wcs and self.wcs != operand.wcs:
+                if isinstance(self.wcs, WCS) and isinstance(operand.wcs, WCS):
+                    log.info("Astropy.WCS does not allow equality checks.")
+                else:
+                    log.info("WCS is not equal.")
```

Given that the user can choose to compare WCS or not, I think an exception be raised here instead of warning if the two don't match.